cmake/deps-helper.cmake

```diff
@@ -0,0 +1,46 @@
+# where to build dependencies
+set(DEPS_PREFIX "${CMAKE_BINARY_DIR}/deps")
+# base path to installed dependencies
+set(DEPS_BASE "${CMAKE_BINARY_DIR}/deps")
+# search path hint for find_X family
+list(APPEND CMAKE_PREFIX_PATH "${DEPS_BASE}/deps/")
+list(REMOVE_DUPLICATES CMAKE_PREFIX_PATH)
+# cmake wants include dirs to exist when a target is created
+file(MAKE_DIRECTORY "${DEPS_BASE}/include/")
```

Comment:
Not sure if I understand why this is needed. What issue did you run into?

Reply:
Good catch. This is a leftover from when the external projects were still living outside of the Find scripts.

Reply:
I was actually referring  to the MAKE_DIRECTORY line :)

Reply:
Ah. Well, the issue is what the comment says: when you create a target (at configure time) with an include dir that does not exist yet (because it's only created at build time when the first dependency is actually installed) cmake complains.

Reply:
You mean because of e.g. this: https://github.com/CVC4/CVC4/pull/6202/files#diff-ab64debc440f89def9a126261fa321af0b17b28764660020f3b9f0934559cd0dR96 ?

Reply:
Exactly.

Reply:
Ok good, then how about changing the comment to:

```
# CMake wants directories specified via INTERFACE_INCLUDE_DIRECTORIES to exist when target property is set.
```